onnxruntime/core/providers/cpu/nn/batch_norm_helper.h

```diff
@@ -122,13 +129,19 @@ class BatchNormHelper {
 
     auto rank = x_shape.NumDimensions();
     auto num_samples = rank > 0 ? orig_dims[0] : 1;  // NCHW
-    auto num_channels = rank > 1 ? orig_dims[1] : 1;
-    auto height = rank > 2 ? orig_dims[2] : 1;
+    const size_t channel_dim = is_nhwc ? rank - 1 : 1;
+    const size_t height_dim = is_nhwc ? 1 : 2;
```

Comment:
When we see 4-D input, isn't orig_dims.size() == 4 and if so don't we bail out in the previous if ? How do we enter the logic where you do the re-arrangement of the dims ? If I understand correctly, we only enter that when the data rank is < 4 and we make it 4-D virtually so that we can use CUDA/CuDNN

Reply:
EDIT: I think I got it now - whether we get NHWC or NCHW data, as long as we get 4-D input we can bail out wihthout creating a virtual new dims